### PR TITLE
[BEAM-13724] [Playground] Get the default example on the frontend

### PR DESCRIPTION
**Pull request by miamihotline**
Made default examples loaded by a separate method.
------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/validate-runner/README.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: playground/frontend/lib/modules/examples/repositories/example_repository.dart [hunk 1]
```diff
@@ -36,6 +37,13 @@ class ExampleRepository {
     return result.categories;
   }
 
+  Future<ExampleModel> getDefaultExample(
+    GetExampleRequestWrapper request,
+  ) async {
+    final result = await _client.getDefaultExample(request);
+    return result.example;
```
Comment (KonradJanica) [inline playground/frontend/lib/modules/examples/repositories/example_repository.dart:44]: Merge into one line.

File: playground/frontend/lib/modules/sdk/components/sdk_selector.dart [hunk 2]
```diff
@@ -67,15 +67,7 @@ class SDKSelector extends StatelessWidget {
...
-                      setExample(
-                        state.defaultExamplesMap![value] ??
-                            ExampleModel(
-                              name: kEmptyExampleName,
-                              path: '',
-                              description: '',
-                              type: ExampleType.example,
-                            ),
-                      );
+                      setExample(state.defaultExamplesMap![value]!);
```
Comment (KonradJanica) [inline playground/frontend/lib/modules/sdk/components/sdk_selector.dart:70]: These exclamation marks are a code smell. Can they be fixed by added asserts or default initializing?

File: playground/frontend/lib/pages/playground/components/playground_page_providers.dart [hunk 3]
```diff
@@ -95,17 +94,27 @@ class PlaygroundPageProviders extends StatelessWidget {
     PlaygroundState playground,
   ) {
     final examplePath = Uri.base.queryParameters[kExampleParam];
+
+    if (exampleState.defaultExamplesMap == null) {
+      exampleState.loadDefaultExamples();
+    }
+
+    if (examplePath == null || examplePath == '') {
```
Comment (KonradJanica) [inline playground/frontend/lib/pages/playground/components/playground_page_providers.dart:102]: if (!examplePath.isEmpty) {

File: playground/frontend/lib/pages/playground/states/examples_state.dart [hunk 4]
```diff
@@ -82,31 +87,41 @@ class ExampleState with ChangeNotifier {
...
   changeSelectorVisibility() {
     isSelectorOpened = !isSelectorOpened;
     notifyListeners();
   }
 
-  _loadDefaultExamples(sdkCategories) async {
+  loadDefaultExamples() async {
+    if (defaultExamplesMap != null) {
+      return;
+    }
```
Comment (KonradJanica) [inline playground/frontend/lib/pages/playground/states/examples_state.dart:106]: Nit: add a new line after the return block.

File: playground/frontend/lib/pages/playground/states/examples_state.dart [hunk 5]
```diff
@@ -82,31 +87,41 @@ class ExampleState with ChangeNotifier {
...
-        entries.add(MapEntry(sdk, defaultExample));
-      }
+    List<MapEntry<SDK, ExampleModel>> defaultExamples = [];
+
+    for (var value in SDK.values) {
+      defaultExamples.add(
+        MapEntry(
+            value,
+            await _exampleRepository.getDefaultExample(
+              GetExampleRequestWrapper('', value),
```
Comment (KonradJanica) [inline playground/frontend/lib/pages/playground/states/examples_state.dart:115]: Please explain why the first parameter is an empty string using code comments.

File: playground/frontend/lib/pages/playground/states/examples_state.dart [hunk 6]
```diff
@@ -82,31 +87,41 @@ class ExampleState with ChangeNotifier {
...
+            )),
+      );
+    }
+
+    defaultExamplesMap!.addEntries(defaultExamples);
```
Comment (KonradJanica) [inline playground/frontend/lib/pages/playground/states/examples_state.dart:120]: Is this exclamation mark is needed? The object is initialized on line 108.

File: playground/frontend/lib/pages/playground/states/examples_state.dart [hunk 7]
```diff
@@ -82,31 +87,41 @@ class ExampleState with ChangeNotifier {
...
+    for (var entry in defaultExamplesMap!.entries) {
```
Comment (KonradJanica) [inline playground/frontend/lib/pages/playground/states/examples_state.dart:121]: Is this exclamation mark is needed? The object is initialized on line 108.

File: playground/frontend/lib/pages/playground/states/examples_state.dart [hunk 8]
```diff
@@ -82,31 +87,41 @@ class ExampleState with ChangeNotifier {
...
+      loadExampleInfo(entry.value, entry.key)
+          .then((value) => defaultExamplesMap![entry.key] = value);
```
Comment (KonradJanica) [inline playground/frontend/lib/pages/playground/states/examples_state.dart:123]: Is this exclamation mark needed? The object is initialized on line 108.

File: playground/frontend/lib/modules/sdk/components/sdk_selector.dart (hunk 2 above)
Comment (miamihotline) [reply playground/frontend/lib/modules/sdk/components/sdk_selector.dart:70]: Hello @KonradJanica! Thank you for the review! I'll fix this in the next commit.

File: playground/frontend/lib/pages/playground/components/playground_page_providers.dart (hunk 3 above)
Comment (miamihotline) [reply playground/frontend/lib/pages/playground/components/playground_page_providers.dart:102]: I'll fix this line in the next commit.

File: playground/frontend/lib/pages/playground/states/examples_state.dart (hunk 4 above)
Comment (miamihotline) [reply playground/frontend/lib/pages/playground/states/examples_state.dart:106]: Sure, I'll add a new line in the next commit.

File: playground/frontend/lib/pages/playground/states/examples_state.dart (hunk 5 above)
Comment (miamihotline) [reply playground/frontend/lib/pages/playground/states/examples_state.dart:115]: Okay, I'll write a comment about this.

File: playground/frontend/lib/pages/playground/states/examples_state.dart (hunk 6 above)
Comment (miamihotline) [reply playground/frontend/lib/pages/playground/states/examples_state.dart:120]: I'll remove it, thank you!

File: playground/frontend/lib/pages/playground/states/examples_state.dart (hunk 8 above)
Comment (miamihotline) [reply playground/frontend/lib/pages/playground/states/examples_state.dart:123]: I'll remove it, thank you!

**Review by KonradJanica — APPROVED** [review]

**Comment by pabloem** [comment]
lgtm thanks all!

**Pull request merged**